### PR TITLE
Improve error messages when schema validation fails

### LINKED ISSUE #5608: Error messages not useful when there are extraneous properties
## Feature Suggestion
Improve error messages when schema validation fails

## Possible Implementation
Update the ajvCompiledJsonSchemaValidator method to include the info on the error.params object when throwing a new error.

## Context
When registering a component it is really hard to figure out what is wrong with the yaml file. This will improve the experience when registering components.

### PR DESCRIPTION
**Pull request by brethubbard**
Signed-off-by: Bret Hubbard <hubbard.bret@gmail.com>

## Hey, I just made a Pull Request!
Improve error messages when schema validation fails, fixes #5608 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/catalog-model/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # @backstage/catalog-model
 
+## 0.7.9
+
+### Patch Changes
+
+- Improve error messages when schema validation fails
```
Comment (adamdmharvey) [inline packages/catalog-model/CHANGELOG.md:7]: No need to directly edit the CHANGELOG itself.  Instead, you can use the changeset functionality.  More info here:
https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets

Comment (brethubbard) [reply packages/catalog-model/CHANGELOG.md]: Ah thank you, I have made that change.

**Review by benjdlambert — APPROVED** [review]

**Comment by benjdlambert** [comment]
Hey @brethubbard, looks like the dco check is failing, you need to sign off your commits.

You can look in here for some details on what to do to make it happen https://github.com/backstage/backstage/pull/5610/checks?check_run_id=2522976187

**Comment by brethubbard** [comment]
Hey @benjdlambert, I knew I missed something in my last commit. It's fixed now, thank you!

**Pull request merged**